core/src/components/input-otp/input-otp.tsx

```diff
@@ -0,0 +1,786 @@
+import type { ComponentInterface, EventEmitter } from '@stencil/core';
+import { Component, Element, Event, Fragment, Host, Prop, State, h, Watch } from '@stencil/core';
+import { printIonWarning } from '@utils/logging';
+import { isRTL } from '@utils/rtl';
+import { createColorClasses } from '@utils/theme';
+import { Method } from 'ionicons/dist/types/stencil-public-runtime';
+
+import { getIonMode } from '../../global/ionic-global';
+import type { Color } from '../../interface';
+
+import type {
+  InputOtpChangeEventDetail,
+  InputOtpCompleteEventDetail,
+  InputOtpInputEventDetail,
+} from './input-otp-interface';
+
+@Component({
+  tag: 'ion-input-otp',
+  styleUrls: {
+    ios: 'input-otp.ios.scss',
+    md: 'input-otp.md.scss',
+  },
+  scoped: true,
+})
+export class InputOTP implements ComponentInterface {
+  private inputRefs: HTMLInputElement[] = [];
+  private inputId = `ion-input-otp-${inputIds++}`;
+  private parsedSeparators: number[] = [];
+
+  /**
+   * The value of the OTP input when it is focused.
```

Comment:
```suggestion
   * Stores the initial value of the input when it receives focus.
   * Used to determine if the value changed during the focus session
   * to avoid emitting unnecessary change events on blur.
```